pkg/daemon/file_writers.go

```diff
@@ -45,12 +46,80 @@ func noOrigFileStampName(fpath string) string {
 	return filepath.Join(noOrigParentDir(), fpath+".mcdnoorig")
 }
 
+func isFileOwnedByRPMPkg(fpath string) (bool, bool, error) {
+	// The first bool returned indicated whether the underlying OS is Mac
+	// The Second bool returned indicated whether the file is owned by the RPM package
+
+	// Check whether the underlying OS is Linux, skip the rest and return if not (e.g. Mac -> no rpm)
+	// If we cannot find the rpm binary, return an error.
+	path, err := exec.LookPath("rpm")
+	if err != nil {
+		return true, false, err
+	}
+
+	// Check whether the given path is owned by an rpm pkg
+	cmd := exec.Command(path, "-qf", fpath)
+	out, err := cmd.CombinedOutput()
+
+	if err == nil && cmd.ProcessState.ExitCode() == 0 {
+		return false, true, nil
+	}
+	fileNotOwnedMsg := fmt.Sprintf("file %s is not owned by any package", fpath)
+	if strings.Contains(string(out), fileNotOwnedMsg) {
+		return false, false, nil
+	}
+	return false, false, fmt.Errorf("command %q returned with unexpected error: %s: %w", cmd, string(out), err)
+}
+
 func createOrigFile(fromPath, fpath string) error {
-	if _, err := os.Stat(noOrigFileStampName(fpath)); err == nil {
-		// we already created the no orig file for this default file
-		return nil
+	orig := false
+
+	// https://issues.redhat.com/browse/OCPBUGS-11437
+	// MCO keeps the pull secret to .orig file once it replaced
+	// Adapt a check used in function deleteStaleData for backwards compatibility: basically if the file doesn't
+	// exist in /usr/etc (on FCOS/RHCOS) and no rpm is claiming it, we assume the orig file doesn't need to be created.
+	// Only do the check for existing files because files that wasn't present on disk before MCD took over will never
+	// be files that were shipped _with_ the underlying OS (e.g. a default chrony config).
+	if _, err := os.Stat(fpath); err == nil {
+		isMac, isOwned, err := isFileOwnedByRPMPkg(fpath)
+		if isOwned {
+			// File is owned by an rpm
+			orig = true
+		} else if !isOwned && (err == nil) {
+			// Run on Linux - check whether the file exist in /usr/etc (on FCOS/RHCOS)
+			if strings.HasPrefix(fpath, "/etc") {
+				if _, err := os.Stat(withUsrPath(fpath)); err != nil {
+					if !os.IsNotExist(err) {
+						return err
+					}
+				} else {
+					orig = true
+				}
+			}
+		} else if isMac {
+			// Run on Mac
+			klog.Infof("Running on Mac,skip orig file preservation.")
```

Comment:
nit: same as above, I think rephrase to "rpm binary not found", and change the references from Mac to "rpmNotFound" or similar, I think that's more clear in case someone gets this error on a non-fedora/rhel machine for some reason